server/src/com/thoughtworks/go/server/service/ConsoleService.java

```diff
@@ -116,7 +117,7 @@ public boolean updateConsoleLog(File dest, InputStream in) throws IOException {
 
         LOGGER.trace("Updating console log [" + dest.getAbsolutePath() + "]");
         try (BufferedWriter writer = new BufferedWriter(new FileWriter(dest, dest.exists()))) {
-            IOUtils.copy(in, writer);
+            IOUtils.copy(in, writer, Charset.defaultCharset());
```

Comment:
This assumes that `Charset.defaultCharset()` on agent and server is identical, which could have other issues when the encodings are not identical (or compatible)

Instead of using a `Writer` to write out the `InputStream`, it'd probably be better if we open `dest` as a (buffered) `FileOutputStream` and write to it instead, so as to do away with gotchas with encodings.


```java
try (OutputStream out = new BufferedOutputStream(new FileOutputStream(dest, dest.exists()))) {
  IOUtils.copy(in, out);
}
```